android/src/main/java/com/rnmaps/maps/MapMarker.java

```diff
@@ -430,6 +430,7 @@ public Object getFeature() {
   public void addToMap(Object collection) {
     MarkerManager.Collection markerCollection = (MarkerManager.Collection) collection;
     marker = markerCollection.addMarker(getMarkerOptions());
+    marker.setTag(this);
```

Comment:
@MartinaMuscarella this code is way too heavy compared to the functionality we're trying to achieve here 

specifically, you're storing a full view collection as an "attachment" to each Marker, and according to google docs:
https://developers.google.com/android/reference/com/google/android/gms/maps/model/Marker#setTag(java.lang.Object)
there is no automated mechanism by google to free memory here

if you look at setOnMarkerClickListener inside MapView.java you can see that only the markerIdentifier is needed.

and if you look at almost all the calls for getMarkerMap in MapView.java you find out only the viewID is needed to dispatch the event through MapManager.receiveEvent